### PR TITLE
Expand CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by 2-4601**
#5082 reminded me of this that's been lying on my hard disk for too long now so I'm just gonna throw it here.

This addresses some of the things that come up all the time at this repository. Dunno if the wiki pages would suit some sections better. The best part of CONTRIBUTING.md is that every time some one opens a new issue or a PR there's a banner:

**"Please review the [guidelines for contributing](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this repository."**

Another question is of course how many people actually read it and follow the link. :smile: 

There's some overlap with README.md but I think you can't repeat these things enough.

I also had references (links to comments in issues and PRs) for almost all parts but decided to just remove them. I don't want to :boom: this more than necessary.

Edit: This is meant to be freebie. Failed at reading my own writing. #youhadonejob

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,33 +1,86 @@
...
+
+#### If it's open it's tracked
+Have you followed all the points in the [Submitting useful bug reports guide](https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports) but nobody has commented on your issue? Is there no milestone or person assigned to it? Don't worry, the developers read every issue and if it's open it means it's tracked and taken into account. It might just take time as other issues have higher priorities. And remember that this is an open source project: Anyone is encouraged to take an active role in fixing open issues.
+
+### Closed issues
+
+#### "My issue was closed without giving a reason!"
+Please understand that writing detailed explanations every time for every issue someone comes up with takes time. Sometimes a reason has been posted earlier to another related issue which you can search for. It's also possible that your issue was not in line with the guidelines of the project (see especially the [Development Ideology](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md#development-ideology)). Or it was just simply decided that the issue is not something that Signal should do at this time.
```
Comment (samlanning) [inline CONTRIBUTING.md]: Nice! :smirk:

**Comment by samlanning** [comment]
Nice PR, we may want to do similar things for Signal-Desktop and Signal-iOS? But then again this repo does get more contributions than most.

Also you could just amend and force push to add the `FREEBIE` comment :wink:

**Comment by 2-4601** [comment]
Amended (and it felt like sin). :cold_sweat: 

Yeah, some parts are similar across the the Signal repos and it would probably be a good idea to add them somewhere. But I don't know whether README or CONTRIBUTING is the better choice. The CONTRIBUTING.mds at iOS and Desktop repositories seem to be specifically for code contributions.

**Comment by samlanning** [comment]
I think as you outline problems that we've only *really* seen in this repo, and that some of what you say is not coding specific / probably best situated in a primary location outside of one of these repos, it may be best to leave the other repos for now?

**Comment by samlanning** [comment]
...at least until there is some kind of operation to consolidate everything...

**Comment by 2-4601** [comment]
@samlanning I was just browsing the mailing list archives and I see you have recently opened a similar topic there. Don't know if you meant this kind of a solution exactly or something related to all OWS projects. Anyway it's best to continue that discussion there. (Perhaps I'll fail to procrastinate subscribing to the list in the near future:)

What I don't like about this PR is that it does not feel friendly and polite enough. But I'm afraid I can't improve that by myself.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,33 +1,86 @@
...
+### Merging can sometimes take a while
+If your pull request follows all the advice above but still has not been merged it usually means the developers haven't simply had the time to review it yet. We understand that this might feel frustrating. We are sorry!
+
+### Bithub
+Accepted pull requests will be rewarded with Bitcoins! After your pull request has been merged you will automatically receive an email to the address you have specified as your Git commit email. Follow the instructions in the email to claim your coins. If you wish to submit your contribution for free please add the word `FREEBIE` in your Git commit message. You may wish to explore some previously merged commits to see how it all works.
+
+
+## How can I contribute?
+Any one can help by
+- advicing new people about the guidelines of this project
```
Comment (haffenloher) [inline CONTRIBUTING.md]: typo: advising

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,33 +1,86 @@
...
+ - redirecting support questions to support@whispersystems.org and the [support site](http://support.whispersystems.org)
+ - redirecting relevant discussions to the [mailing list](https://lists.riseup.net/www/info/whispersystems)
+- improving documentation
```
Comment (haffenloher) [inline CONTRIBUTING.md]: readers will probably wonder "where and how?"
I don't really have an answer myself though :\ maybe in the wiki?

**Comment by samlanning** [comment]
@2-4601: I incorrectly assumed you'd seen the message and that this PR was in response to that haha...

Yeah I did mean something more general across all projects though, but this PR certainly doesn't hurt! :smile:

**Comment by samlanning** [comment]
Started development of wiki page: https://github.com/WhisperSystems/Signal-Android/wiki/Welcome-to-the-Signal-Community

(will write to mailing list now)

**Comment by samlanning** [comment]
I stole a lot of your wording :wink: 

You probably want to replace some of what you've written with links into this wiki page now? And maybe help me flesh it out a little :stuck_out_tongue:

**Comment by 2-4601** [comment]
Browsing through the log it seems CONTRIBUTING.md was [originally created](https://github.com/WhisperSystems/Signal-Android/commit/ca561d76ffcb4760b012e34e4144651321ff0e25) by liliakai to stop people from creating issues/prs for translations and to redirect them to Transifex. Then the Development ideology and bug report sections were added and it has grown organically from there. 

My motivation for this is to have some document that can be referred to when someone in this tracker opens a duplicate issue, bumps an issue or asks how to send a picture. I guess one could use canned replies (e.g. with TextExpander) but referring to a documentation section is better.

I read your wiki draft and it is good. It has a lot of information though. Do you plan to move some other wiki pages under there too? For example [Submitting useful bug reports](https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports) wiki page under [Reporting Bugs](https://github.com/WhisperSystems/Signal-Android/wiki/Welcome-to-the-Signal-Community#reporting-bugs) section? The big question is where should all this information go? I think we want to use CONTRIBUTING.md for the great feature that GitHub links to it every time someone opens an issue. But should it just contain links to wiki pages? If it contains links to just one big wiki page then isn't it effectively the same as that page? :) Do links create additional hoops for users to go through to read docs (which they don't want to read in the first place:)? But the wiki pages are actually quite nice because they can be quickly updated.

In the mean time I'll just address @haffenloher's comments, thanks.

**Pull request closed without merging**